airbyte-featureflag/src/main/kotlin/Client.kt

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.featureflag
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import com.launchdarkly.sdk.ContextKind
+import com.launchdarkly.sdk.LDContext
+import com.launchdarkly.sdk.LDUser
+import com.launchdarkly.sdk.server.LDClient
+import java.lang.Thread.MIN_PRIORITY
+import java.nio.file.Path
+import java.nio.file.StandardWatchEventKinds
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.thread
+import kotlin.concurrent.write
+import kotlin.io.path.isRegularFile
+
+/**
+ * Open Questions:
+ * 1. Can the context and flag be combined into one class?
+ *   - i.e. Does it make sense to one a feature-flag apply to two different context types?
+ * 2. Should the default value be merged into the Flag type instead of specified on each check?
+ *   - Or maybe still allow it to be specified, but make it an optional parameter
```

Comment:
what are we trying to achieve with the optional? it seems like whoever declares the flag is well situated to decide what the default is. it's not obvious to me why the person checking it would get any say in what the default is (even optionally).

Reply:
This comment was my open thoughts on an earlier implementation of this code and is no longer applicable.   

The LaunchDarkly SDK has you provide the default response _every time_ you call to check the status of the flag, and my first (rough) approach followed this same design.  It's not a good design, as it's possible that two sections in the code checking the same feature-flag would have two different defaults (if that flag no longer existed or couldn't be found), which is not a situation that should be supported.  I merged this into the Flag definition and it cannot be overwritten outside of the flag definition.